tests/unit/frameworks/vernon/meta_continual_learning_test.py

```diff
@@ -0,0 +1,149 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2020, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+
+import unittest
+
+import torch
+
+from nupic.research.frameworks.continual_learning.maml_utils import clone_model
+from nupic.research.frameworks.vernon import MetaContinualLearningExperiment
+
+# Retrieve function that updates params in place.
+# This enables taking gradients of gradients.
+update_params = MetaContinualLearningExperiment.update_params
+
+
+class Quadractic(torch.nn.Module):
+    """Quadratic layer: Computes W^T W x"""
+    def __init__(self):
+        super().__init__()
+        self.weight = torch.nn.Parameter(torch.tensor([
+            [0.94, 0.07],
+            [0.40, 0.21]
+        ]))
+
+    def forward(self, x):
+        """Compute W^T W x"""
+        out = torch.matmul(self.weight, x)
+        out = torch.matmul(self.weight.transpose(1, 0), out)
+        return out
+
+
+class GradsOfGradsTest(unittest.TestCase):
+    """
+    Perform tests for taking gradients of gradients. Specifically, this uses a
+    quadratic layer as a test example since it yields non-trivial results (unlike a
+    linear layer) and enables hand derived expected values.
+    """
+
+    def setUp(self):
+        # Input passed to quadratic function: a W^T W x
+        self.left_input = torch.tensor([[1., 1.]], requires_grad=True)  # a
+        self.right_input = torch.tensor([[0.32], [0.72]], requires_grad=True)  # x
+
+        # Expected gradient of gradient for quadratic layer updated via SGD.
+        self.expected_grad = torch.tensor([[0.3972, 0.6762], [0.2869, 0.4458]])
+
+    def test_auto_grad_with_quadratic_function(self):
+        """
+        Test use of pytorch's autograd to keep track of gradients of gradients.
```

Comment:
There are some formulas written as comments here in this function. If there is an external reference, maybe add the link here or further explain it. 

It is just a test, so completely optional, but it would help someone maintaining the code in the future.

Reply:
Good point. I included the derivations in the comments.